### ISSUE TITLE
`pgpainless-cli inline-verify` emits output on stdout even when no valid signatures are found

### ISSUE THREAD

**Issue by dkg** [vanitasvitae/pgpainless#459]
The SOP spec explicitly says:

> If no valid signatures are found, `sop inline-verify` fails with `NO_SIGNATURE` and emits nothing on standard output.

But `pgpainless-cli` emits the message on stdout even if it doesn't find a valid signature:

```
0 dkg@bob:$ pgpainless-cli version --extended
PGPainless-SOP 1.6.8
https://codeberg.org/PGPainless/pgpainless/src/branch/master/pgpainless-sop

Implementation of the Stateless OpenPGP Protocol Version 07
https://datatracker.ietf.org/doc/html/draft-dkg-openpgp-stateless-cli-07

Based on pgpainless-core 1.6.8
https://pgpainless.org

Using Bouncy Castle 1.77
https://www.bouncycastle.org/java.html
0 dkg@bob:$ pgpainless-cli generate-key alice > alice.key
0 dkg@bob:$ pgpainless-cli generate-key bob > bob.key
0 dkg@bob:$ pgpainless-cli extract-cert < alice.key > alice.cert
0 dkg@bob:$ pgpainless-cli extract-cert < bob.key > bob.cert
0 dkg@bob:$ echo test | pgpainless-cli inline-sign alice.key > test.msg.signed
0 dkg@bob:$ pgpainless-cli inline-verify alice.cert < test.msg.signed 
test
0 dkg@bob:$ pgpainless-cli inline-verify bob.cert < test.msg.signed 
test
No verifiable signature found.
3 dkg@bob:$ 
```

This is likely to produce a security problem if a `sop` consumer (or `sopv` consumer) depends on the specification's stdout declaration and ignores the return code.

**Comment by dkg** [comment]
this appears to be the case for CSF messages as well as "normal" inline-signed messages.

**Comment by vanitasvitae** [comment]
Ah yes, PGPainless currently doesn't cache the message content, but instead emits it directly.
Originally, I decided to do so due to performance reasons, but maybe it's time to change that. I'm currently working on a decryption/verification API in BC, so I will take this feedback into account there and make it hold back the decryption result by default (with the option to turn on streaming).
For the mean time, I could create a stream-wrapper in the SOP library until PGPainless 2 will then switch to using the new BC API.

**Comment by dkg** [comment]
I am realizing you said "decryption result" above here, and i agree with you that pgpainless-cli should not emit cleartext until the decryption is confirmed.  But just to clarify -- _this_ ticket is not about decryption, but rather about signature verification for inline-signed messages,  right?

**Comment by vanitasvitae** [comment]
Yeah, PGPainless implements decryption and verification with the same API, so the returned result object is the same.
